packages/react/src/components/Tag/Tag.js

```diff
@@ -50,40 +51,67 @@ const Tag = ({
       event.stopPropagation();
       onClose(event);
     }
-  };
-  return filter ? (
-    <div
-      className={tagClasses}
-      aria-label={
-        title !== undefined
-          ? `${title} ${children}`
-          : `Clear filter ${children}`
-      }
-      id={tagId}
-      {...other}>
+  }
+
+  if(filter) {
+    return (
+      <div
+        className={tagClasses}
+        aria-label={
+          title !== undefined
+            ? `${title} ${children}`
+            : `Clear filter ${children}`
+        }
+        id={tagId}
+        {...other}>
+        <span
+          className={`${prefix}--tag__label`}
+          title={typeof children === 'string' ? children : null}>
+          {children !== null && children !== undefined ? children : TYPES[type]}
+        </span>
+        <button
+          type="button"
+          className={`${prefix}--tag__close-icon`}
+          onClick={handleClose}
+          disabled={disabled}
+          aria-labelledby={tagId}
+          title={title}>
+          <Close16 />
+        </button>
+      </div>
+    );
+  } else if(CustomIconElement) {
+    return (
+      <div
+        className={tagClasses}
+        aria-label={
+          title !== undefined
+            ? `${title} ${children}`
+            : `Clear filter ${children}`
+        }
+        id={tagId}
+        {...other}>
+        <button
+          className={`${prefix}--tag__custom-icon`}>
+          <CustomIconElement />
+        </button>
```

Comment:
```suggestion
        <button
          type="button"
          className={`${prefix}--tag__custom-icon`}>
          <CustomIconElement />
        </button>
```

I believe this also needs a `type` attribute

Reply:
Hmm, thinking about it, the custom icon isn't a button in this context (it can't be clicked). Closing the `CustomIconElement` in a `<div>` would provide the same effect, so I'll be changing it to that instead.